packages/graphql-codegen-cli/src/execute-plugin.ts

```diff
@@ -104,5 +104,7 @@ export async function executePlugin(options: ExecutePluginOptions, pluginPackage
     }
   }
 
-  return pluginPackage.plugin(outputSchema, options.documents, options.config);
+  return pluginPackage.plugin(outputSchema, options.documents, options.config, {
+    outputFile: options.outputFilename
+  });
```

Comment:
@dotansimha I added 4th argument to PluginFunction so plugin can tell what is the output format. Some folks might want to use introspection result as a `.json` file, others directly in TypeScript without having another build step to move the `.json`.